### PR TITLE
Create formatter scaffolding

### PR DESCRIPTION
**Pull request by difince**
FormatWriter interface added
The logic responsible for parsing and executing the template is in
Formatter.Write method
`command list` use it

Signed-off-by: Diana Atanasova <dianaa@vmware.com>

### CODE REVIEW

File: cmd/command/list/list.go [hunk 1]
```diff
@@ -16,12 +16,11 @@ package list
 import (
 	"github.com/spf13/viper"
 	"html/template"
-	"io"
 	"strings"
-	"text/tabwriter"
 
 	"github.com/edgexfoundry-holding/edgex-cli/config"
 	request "github.com/edgexfoundry-holding/edgex-cli/pkg"
+	"github.com/edgexfoundry-holding/edgex-cli/pkg/formater"
```
Comment (mkbhanda) [inline cmd/command/list/list.go:23]: s/formater/formatter
the spelling is too close

File: cmd/command/list/list.go [hunk 2]
```diff
@@ -61,21 +60,18 @@ func listHandler(cmd *cobra.Command, args []string) (err error) {
...
-	if err != nil {
-		return err
-	}
-	w.Flush()
+	formatter := newCmdListFormatter()
+	err = formatter.Write(responses)
 	return
 }
 
+func newCmdListFormatter() formater.FormatWriter {
```
Comment (mkbhanda) [inline cmd/command/list/list.go:68]: s/formater/formatter

File: cmd/command/list/list.go [hunk 3]
```diff
@@ -61,21 +60,18 @@ func listHandler(cmd *cobra.Command, args []string) (err error) {
...
+	if viper.GetBool("verbose") {
+		return &formater.EmptyFormatter{}
```
Comment (mkbhanda) [inline cmd/command/list/list.go:70]: ditto

File: pkg/formater/formatter.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
+package formater
```
Comment (mkbhanda) [inline pkg/formater/formatter.go:1]: ditto

File: cmd/command/list/list.go (hunk 1 above)
Comment (difince) [reply cmd/command/list/list.go:23]: I will remember it :)

**Comment by difince** [comment]
comments addressed

**Pull request merged**